dd-sdk-android/src/main/kotlin/com/datadog/android/v2/core/internal/storage/ConsentAwareStorage.kt

```diff
@@ -41,24 +49,34 @@ internal class ConsentAwareStorage(
             TrackingConsent.NOT_GRANTED -> null
         }
 
+        val batchFile = orchestrator?.getWritableFile()
+        val metadataFile = if (batchFile != null) batchMetadataFileProvider(batchFile) else null
+
         val writer = object : BatchWriter {
             @WorkerThread
             override fun currentMetadata(): ByteArray? {
-                // TODO RUMM-2186 handle writing/updating batch metadata in separate file
-                return null
+                if (orchestrator == null || metadataFile == null) return null
+
+                return batchMetadataFileHandler.readData(metadataFile)
             }
 
             @WorkerThread
-            override fun write(event: ByteArray, eventId: String, newMetadata: ByteArray) {
+            override fun write(event: ByteArray, eventId: String, newMetadata: ByteArray?) {
                 // prevent useless operation for empty event / null orchestrator
                 if (event.isEmpty() || orchestrator == null) {
                     listener.onDataWritten(eventId)
                     return
                 }
 
-                val file = orchestrator.getWritableFile(event.size)
+                if (!checkEventSize(event.size)) {
```

Comment:
Isn't this already checked in the `FileHandler` ?

Reply:
It was moved from `FileOrchestrator` to the `DataWriter`, but `DataWriter` is not used by the `ConsentAwareStorage` right now (and probably won't, in fact in the writer callback we need only `Serializer`, so logically `Storage` replaces `DataWriter`), so adding this check here as well.